src/unix/bsd-ifaddrs.c

```diff
@@ -31,11 +31,17 @@
 #include <net/if_dl.h>
 #endif
 
-static int uv__ifaddr_exclude(struct ifaddrs *ent) {
+static int uv__ifaddr_exclude(struct ifaddrs *ent, int exclude_type) {
   if (!((ent->ifa_flags & IFF_UP) && (ent->ifa_flags & IFF_RUNNING)))
     return 1;
   if (ent->ifa_addr == NULL)
     return 1;
+  /*
+   * Just see whether sa_family equals to AF_LINK or not if exclude_type is
+   * UV__EXCLUDE_IFPHYS
+   */
+  if (exclude_type == UV__EXCLUDE_IFPHYS)
+    return (ent->ifa_addr->sa_family != AF_LINK);
 #if defined(__APPLE__) || defined(__FreeBSD__) || defined(__DragonFly__)
   /*
    * On BSD getifaddrs returns information related to the raw underlying
```

Comment:
If `UV__EXCLUDE_IFPHYS`, then only returns `(ent->ifa_addr->sa_family != AF_LINK)`.

Otherwise, it depends on system. If it's OSX / FreeBSD / DragonFly then it depends on `AF_LINK`, or it's depends on `PF_INET`.